src/core/server/plugins/plugins_system.ts

```diff
@@ -158,13 +158,41 @@ export class PluginsSystem {
         timeout: 30 * Sec,
         errorMessage: `Start lifecycle of "${pluginName}" plugin wasn't completed in 30sec. Consider disabling the plugin and re-start.`,
       });
-
       contracts.set(pluginName, contract);
     }
+    await this.healthCheckOpenSearchPlugins(deps);
 
     return contracts;
   }
 
+  private async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
+    // make _cat/plugins?format=json call to the OpenSearch instance
+    const opensearchPlugins = await this.getOpensearchPlugins(deps);
+    for (const pluginName of this.satupPlugins) {
+      this.log.debug(`For plugin "${pluginName}"...`);
+      const plugin = this.plugins.get(pluginName)!;
+      const pluginBackendDeps = new Set([...plugin.requiredOpenSearchPlugins]);
```

Comment:
Why do you need a set here? Is it valid for `requiredOpenSearchPlugins` for a particular plugin to include duplicate entries?

Reply:
I don't think so but having set here would avoid any duplicate entries being queried twice with OpenSearch API call.